docs/guides/data-ingestion.mdx

```diff
@@ -0,0 +1,196 @@
+---
+sidebar_position: 2
+title: Data Ingestion
+description: Complete guide on how to ingest data into ReductStore using the  SDKs or HTTP API.
+---
+
+import CodeBlock from "@theme/CodeBlock";
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<head>
+    <link rel="canonical" href="https://www.reduct.store/docs/guides/data-ingestion"/>
+</head>
+
+# Data Ingestion
+
+## Concepts
+
+Data ingestion in ReductStore is based on HTTP API. Each record is sent as a binary object in the body of the POST request
+and must have the following information:
+* Bucket name, which must exist in the instance.
+* Entry name, which is a unique identifier of the record in the bucket. If the entry doesn't exist, it will be created.
+* Timestamp as a Unix timestamp in microseconds, which is used for sorting and querying data. Must be unique for each record.
+
+Additionally, a writer can add the following information:
+
+* Labels as key-value pairs which can be used for annotating and querying data
+* Content type which can be used for data interpretation
+
+ReductStore uses a streaming approach to ingest data. The server receives the data in chunks and streams them in a file system. This enables efficient data ingestion and reduces the memory footprint of the server.
+
+In case of little size records, ReductStore provides a batch ingestion mode. In this mode, the server receives a batch of records in a single request to reduce the overhead of the HTTP protocol.
+
+
+### Limitations
+
+The following limitations are applied to the data ingestion process:
+
+* Currently, records and their metadata are immutable. Once a record is ingested successfully, it cannot be overwritten or changed.
+* ReductStore doesn't have any limitations on the size of the record, however, the metadata is sent in the HTTP headers of the request and the size of the headers can be limited by the server or client configuration.
+* Since ReaductStore pre-allocate space for records and store them together in a block, the size must be known in advance even though the record is sent in chunks.
```

Comment:
I got confused with the "block size" like for the parameter "max block size", but you meant the size of the blob that need to be stored?

Like: "ReaductStore needs to know the total size of a record in advance, even if it's sent in chunks. This allows it to reserve the right amount of space for optimal performance."